packages/invoices/src/OutboundShipment/api/api.ts

```diff
@@ -1,24 +1,71 @@
 import { DraftOutboundLine } from './../../types';
 import {
+  UpdateOutboundShipmentUnallocatedLineInput,
+  InsertOutboundShipmentUnallocatedLineInput,
   DeleteOutboundShipmentLineInput,
   RecordPatch,
   UpdateOutboundShipmentInput,
   InvoiceNodeStatus,
   UpdateOutboundShipmentStatusInput,
   InsertOutboundShipmentLineInput,
   UpdateOutboundShipmentLineInput,
+  InvoiceLineNodeType,
 } from '@openmsupply-client/common';
 import { Invoice, InvoiceLine } from '../../types';
-import { getSdk, InvoiceQuery } from './operations.generated';
+import { getSdk } from './operations.generated';
 
 export type OutboundShipmentApi = ReturnType<typeof getSdk>;
 
-const invoiceGuard = (invoiceQuery: InvoiceQuery) => {
-  if (invoiceQuery.invoice.__typename === 'InvoiceNode') {
-    return invoiceQuery.invoice;
-  }
-
-  throw new Error(invoiceQuery.invoice.error.description);
+const outboundParsers = {
```

Comment:
sorry, the diff is a bit much in here as I moved all the 'parsers' into this object